deps/rabbitmq_stream/src/rabbit_stream_reader.erl

```diff
@@ -75,7 +75,7 @@
          stream_leaders :: #{stream() => pid()},
          stream_subscriptions :: #{stream() => [subscription_id()]},
          credits :: atomics:atomics_ref(),
-         authentication_state :: atom(),
+         authentication_state :: any(),
```

Comment:
Additionally, it would be nice to add a new opaque type `rabbit_auth_mechanism:state()` which is returned from `init/1`, and accepted by and returned from `handle_response/2` callbacks.